packages/backend/src/db/task/taskStore.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import * as mongoose from 'mongoose';
+import { Model } from 'mongoose';
+import { Task, TaskDocument, TaskModel } from './task.schema';
+
+@Injectable()
+export class TaskStoreService {
+  constructor(
+    @InjectModel(Task.name)
+    private readonly taskModel: Model<TaskDocument>,
+  ) {}
+
+  async create(createdTaskModel: TaskModel): Promise<TaskDocument> {
+    return this.taskModel.create(createdTaskModel);
+  }
+
+  async findOne(id: string | mongoose.Types.ObjectId): Promise<TaskDocument> {
+    return this.taskModel.findOne({ _id: id }).exec();
+  }
+
+  async findAll(): Promise<TaskDocument[]> {
+    return this.taskModel.find().exec();
+  }
+
+  async update(
+    id: string,
+    updateTaskModel: Partial<TaskModel>,
+  ): Promise<TaskDocument> {
+    return this.taskModel.findOneAndUpdate({ _id: id }, updateTaskModel).exec();
```

Comment:
By default, the `findOneAndUpdate()` method will return the TaskDocument **BEFORE** the update was applied. I would suggest to add in a third parameter `{new: true}` to allow returning latest changed TaskDocument. 

i.e. 

```ts
return this.taskModel.findOneAndUpdate({ _id: id }, updateTaskModel, {new: true}).exec();
```

More info please see this [link](https://mongoosejs.com/docs/tutorials/findoneandupdate.html)

Reply:
Thanks for this, I see what you are getting at. However, this is not relevant to our use case. When we update the tasks, at the update endpoints, we do not return the updated document. The only other place where update is used is in the two GET endpoints where the recurrence rules are checked. Not all tasks will get updated because of these rules, so we have to fetch all the tasks for the house/user again anyway.